docs/architecture/single-controller-rayon-design.md

```
@@ -0,0 +1,501 @@
+    let mut decoder = EntityDecoder::with_cache(content, &entity_index_arc, &mut cache);
+    // ... per-entity work ...
+});
+```
+
+This eliminates the ~few ms per-call FxHashMap allocation, retains decoder cache hits across consecutive rayon tasks on the same thread, and has zero contention because each thread has its own.
+
+### 5.4 SAB ring buffer for streaming output (`packages/geometry/src/mesh-ring-buffer.ts`)
+
+The canonical pattern for streaming results out of rayon ([reference](https://github.com/PaulKinlan/sab-ring-buffer)):
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)